mplib/urdf_utils.py

```diff
@@ -0,0 +1,188 @@
+                "reason: Default (collides at all-zeros qpos)"
+            )
+        acm.set_entry(link1, link2, True)
+        _ = ET.SubElement(
+            root,
+            "disable_collisions",
+            attrib={"link1": link1, "link2": link2, "reason": "Default"},
+        )
+
+    # 3. disable collision pairs that always collide and never collide via sampling
+    n_links = len(user_link_names)
+    collision_cnt = np.zeros((n_links, n_links), dtype=int)
+    for _ in range(num_samples):
+        robot.set_qpos(pinocchio_model.get_random_configuration(), True)
+        for collision in fcl_model.check_self_collision():
+            u = link_name_2_idx[collision.link_name1]
+            v = link_name_2_idx[collision.link_name2]
+            collision_cnt[u][v] += 1
+
+    for i, link1 in enumerate(user_link_names):
+        for j in range(i + 1, n_links):
+            link2 = user_link_names[j]
+            if acm.get_entry(link1, link2) is not None:  # already ignored
+                continue
+            if (cnt := (collision_cnt[i][j] + collision_cnt[j][i])) == num_samples:
+                if verbose:
+                    print(
+                        f"Ignore collision pair: ({link1}, {link2}), "
+                        "reason: Always collide"
+                    )
+                _ = ET.SubElement(
+                    root,
+                    "disable_collisions",
+                    attrib={"link1": link1, "link2": link2, "reason": "Always"},
+                )
+            elif cnt == 0:
+                if verbose:
+                    print(
+                        f"Ignore collision pair: ({link1}, {link2}), "
+                        "reason: Never collide"
+                    )
+                _ = ET.SubElement(
+                    root,
+                    "disable_collisions",
+                    attrib={"link1": link1, "link2": link2, "reason": "Never"},
+                )
+
+    return minidom.parseString(ET.tostring(root)).toprettyxml(indent="  ")
+
+
+def replace_urdf_package_keyword(
+    urdf_path: str | Path,
+    new_package_keyword: str = "",
+    *,
+    save_path: Optional[str | Path] = None,
+) -> Path:
+    """
+    Some ROS URDF files use package:// keyword to refer the package dir.
+    Replace it with the given string (default is empty)
+
+    :param urdf_path: Path to a Unified Robot Description Format file.
+    :param new_package_keyword: the string to replace 'package://' keyword
+    :param save_path: Path to save the modified URDF file.
+        If ``None``, create a temporary file.
+    :return: Path to the modified URDF file
+    """
+    if save_path is None:
+        _, save_path = tempfile.mkstemp(suffix=".urdf")
+    save_path = Path(save_path)
+    if not (save_dir := save_path.parent).is_dir():
+        save_dir.mkdir(parents=True)
+
+    with Path(urdf_path).open("r") as in_f:
+        if "package://" in (content := in_f.read()):
```

Comment:
If there is no package keyword, the content will be empty and so is the output urdf file